lib/DiscoveryStreamFeed.jsm

```diff
@@ -1164,20 +1169,29 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
   }
 
   recordCampaignImpression(campaignId) {
-    let impressions = this.readImpressionsPref(PREF_SPOC_IMPRESSIONS);
+    let impressions = this.readDataPref(PREF_SPOC_IMPRESSIONS);
 
     const timeStamps = impressions[campaignId] || [];
     timeStamps.push(Date.now());
     impressions = { ...impressions, [campaignId]: timeStamps };
 
-    this.writeImpressionsPref(PREF_SPOC_IMPRESSIONS, impressions);
+    this.writeDataPref(PREF_SPOC_IMPRESSIONS, impressions);
   }
 
   recordTopRecImpressions(recId) {
-    let impressions = this.readImpressionsPref(PREF_REC_IMPRESSIONS);
+    let impressions = this.readDataPref(PREF_REC_IMPRESSIONS);
     if (!impressions[recId]) {
       impressions = { ...impressions, [recId]: Date.now() };
-      this.writeImpressionsPref(PREF_REC_IMPRESSIONS, impressions);
+      this.writeDataPref(PREF_REC_IMPRESSIONS, impressions);
+    }
+  }
+
+  recordBlockCampaignId(campaignId) {
+    const campaigns = this.readDataPref(PREF_CAMPAIGN_BLOCKS);
+    // This is a number, not a string. TODO
```

Comment:
Is this a TODO for this ticket? If not is there a follow-on filed?

Reply:
Ah ha, yes, that was a todo for me to follow up on. I put it there as a contingency plan in case I forgot, now it's in the pr and is more likely to be caught in the review process.

Example, if I'm working on a bit of code, and 2 or 3 ideas or concerns are noticed, I quickly leave these sort of todo notes, and I usually catch them before pushing the code up, so ideally they never get seen, until I forget. And this way it's I'm likely to get a reminder. :D